packages/discord.js/src/structures/InteractionCollector.js

```diff
@@ -40,7 +40,7 @@ class InteractionCollector extends Collector {
      * The message from which to collect interactions, if provided
      * @type {?Snowflake}
      */
-    this.messageId = options.message?.id ?? options.interactionResponse?.interaction.message?.id ?? null;
```

Comment:
Removing this condition will lead to invalid interaction collected when we want to collect component interaction after the call of the "update" function on an ephemeral message
(see my long comment with a POC)

Reply:
Thanks! I suspected there would be an edge case this would miss. Will keep looking into it.